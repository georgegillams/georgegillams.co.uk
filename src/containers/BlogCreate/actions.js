import { createRoutine } from 'redux-saga-routines';

import { CREATE_BLOG } from './constants';

export const createBlog = createRoutine(CREATE_BLOG);
