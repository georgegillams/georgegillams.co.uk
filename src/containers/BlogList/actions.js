import { createRoutine } from 'redux-saga-routines';

import { LOAD_BLOGS, DELETE_BLOG } from './constants';

export const loadBlogs = createRoutine(LOAD_BLOGS);
export const deleteBlog = createRoutine(DELETE_BLOG);
