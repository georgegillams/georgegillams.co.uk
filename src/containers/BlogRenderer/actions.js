import { createRoutine } from 'redux-saga-routines';

import { LOAD_BLOG, UPDATE_BLOG } from './constants';

export const loadBlog = createRoutine(LOAD_BLOG);
export const updateBlog = createRoutine(UPDATE_BLOG);
