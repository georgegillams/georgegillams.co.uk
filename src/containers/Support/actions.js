import { createRoutine } from 'redux-saga-routines';

import { LOAD_LINKS, CREATE_LINK, DELETE_LINK } from './constants';

export const loadLinks = createRoutine(LOAD_LINKS);
export const createLink = createRoutine(CREATE_LINK);
export const deleteLink = createRoutine(DELETE_LINK);
