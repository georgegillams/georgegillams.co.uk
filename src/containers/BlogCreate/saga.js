import { takeLatest, put, call, select } from 'redux-saga/effects';

import request from 'client-utils/common/request';
import apiStructure from 'helpers/common/apiStructure';
import { selectState } from './selectors';

import { createBlog } from './actions';

export function* doCreateBlog() {
  const currentState = yield select(selectState());
  const { blogToCreate } = currentState;
  const requestURL = apiStructure.createBlog.fullPath;

  try {
    yield put(createBlog.request());

    const result = yield call(request, requestURL, {
      method: 'POST',
      body: JSON.stringify(blogToCreate),
      headers: {
        'Content-Type': 'application/json',
      },
    });

    if (result.error) {
      yield put(createBlog.failure(result));
    } else {
      yield put(createBlog.success(result));
      window.location = `/blog/${result.id}`;
    }
  } catch (err) {
    yield put(createBlog.failure(err));
  } finally {
    yield put(createBlog.fulfill());
  }
}

export default function* saga() {
  yield takeLatest(createBlog.TRIGGER, doCreateBlog);
}
