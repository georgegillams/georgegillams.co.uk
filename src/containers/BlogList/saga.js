import { takeLatest, put, call, select } from 'redux-saga/effects';

import request from 'utils/common/request';
import apiStructure from 'helpers/apiStructure';
import { selectState } from './selectors';

import { loadBlogs, deleteBlog } from './actions';

export function* doLoadBlogs() {
  const requestURL = apiStructure.loadBlogs.fullPath;

  try {
    yield put(loadBlogs.request());

    const result = yield call(request, requestURL, {
      method: 'GET',
    });

    if (result.error) {
      yield put(loadBlogs.failure(result));
    } else {
      yield put(loadBlogs.success(result));
    }
  } catch (err) {
    yield put(loadBlogs.failure(err));
  } finally {
    yield put(loadBlogs.fulfill());
  }
}

export function* doDeleteBlog() {
  const currentState = yield select(selectState());
  const { blogToDelete } = currentState;
  const requestURL = apiStructure.deleteBlog.fullPath;

  try {
    yield put(deleteBlog.request());

    const result = yield call(request, requestURL, {
      method: 'POST',
      body: JSON.stringify({ id: blogToDelete }),
      headers: {
        'Content-Type': 'application/json',
      },
    });

    if (result.error) {
      yield put(deleteBlog.failure(result));
    } else {
      yield put(deleteBlog.success());
    }
    yield put(loadBlogs.trigger());
  } catch (err) {
    yield put(deleteBlog.failure(err));
  } finally {
    yield put(deleteBlog.fulfill());
  }
}

export default function* saga() {
  yield takeLatest(loadBlogs.TRIGGER, doLoadBlogs);
  yield takeLatest(deleteBlog.TRIGGER, doDeleteBlog);
}
