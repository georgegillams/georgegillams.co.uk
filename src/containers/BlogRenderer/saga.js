import { takeLatest, put, call, select } from 'redux-saga/effects';

import request from 'utils/common/request';
import apiStructure from 'helpers/apiStructure';

import { selectState } from './selectors';
import { loadBlog, updateBlog } from './actions';

export function* doLoadBlog() {
  const currentState = yield select(selectState());
  const { blogId } = currentState;
  const requestURL = apiStructure.loadBlog.fullPath.split(':id').join(blogId);

  try {
    yield put(loadBlog.request());

    const result = yield call(request, requestURL, {
      method: 'GET',
    });

    if (result.error) {
      yield put(loadBlog.failure(result));
    } else {
      yield put(loadBlog.success(result));
    }
  } catch (err) {
    yield put(loadBlog.failure(err));
  } finally {
    yield put(loadBlog.fulfill());
  }
}

export function* doUpdateBlog() {
  const currentState = yield select(selectState());
  const { blogToUpdate } = currentState;
  const requestURL = apiStructure.updateBlog.fullPath;

  try {
    yield put(updateBlog.request());

    const result = yield call(request, requestURL, {
      method: 'POST',
      body: JSON.stringify(blogToUpdate),
      headers: {
        'Content-Type': 'application/json',
      },
    });

    if (result.error) {
      yield put(updateBlog.failure(result));
    } else {
      yield put(updateBlog.success(result));
    }
    yield put(loadBlog.trigger(blogToUpdate.id));
  } catch (err) {
    yield put(updateBlog.failure(err));
  } finally {
    yield put(updateBlog.fulfill());
  }
}

export default function* saga() {
  yield takeLatest(loadBlog.TRIGGER, doLoadBlog);
  yield takeLatest(updateBlog.TRIGGER, doUpdateBlog);
}
