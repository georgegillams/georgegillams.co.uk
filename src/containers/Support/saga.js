import { takeLatest, put, call, select } from 'redux-saga/effects';

import request from 'utils/common/request';
import apiStructure from 'helpers/apiStructure';

import { selectState } from './selectors';
import { loadLinks, createLink, deleteLink } from './actions';

export function* doLoadLinks() {
  const requestURL = apiStructure.loadSupport.fullPath;

  try {
    yield put(loadLinks.request());

    const result = yield call(request, requestURL, {
      method: 'GET',
    });

    if (result.error) {
      yield put(loadLinks.failure(result));
    } else {
      yield put(loadLinks.success(result));
    }
  } catch (err) {
    yield put(loadLinks.failure(err));
  } finally {
    yield put(loadLinks.fulfill());
  }
}

export function* doCreateLink() {
  const currentState = yield select(selectState());
  const { linkToCreate } = currentState;
  const requestURL = apiStructure.createSupport.fullPath;

  try {
    yield put(createLink.request());

    const result = yield call(request, requestURL, {
      method: 'POST',
      body: JSON.stringify(linkToCreate),
      headers: {
        'Content-Type': 'application/json',
      },
    });

    if (result.error) {
      yield put(createLink.failure(result));
    } else {
      yield put(createLink.success(result));
    }
    yield put(loadLinks.trigger());
  } catch (err) {
    yield put(createLink.failure(err));
  } finally {
    yield put(createLink.fulfill());
  }
}

export function* doDeleteLink() {
  const currentState = yield select(selectState());
  const { linkToDelete } = currentState;
  const requestURL = apiStructure.deleteSupport.fullPath;

  try {
    yield put(deleteLink.request());

    const result = yield call(request, requestURL, {
      method: 'POST',
      body: JSON.stringify(linkToDelete),
      headers: {
        'Content-Type': 'application/json',
      },
    });

    if (result.error) {
      yield put(deleteLink.failure(result));
    } else {
      yield put(deleteLink.success(result));
    }
    yield put(loadLinks.trigger());
  } catch (err) {
    yield put(deleteLink.failure(err));
  } finally {
    yield put(deleteLink.fulfill());
  }
}

export default function* saga() {
  yield takeLatest(loadLinks.TRIGGER, doLoadLinks);
  yield takeLatest(createLink.TRIGGER, doCreateLink);
  yield takeLatest(deleteLink.TRIGGER, doDeleteLink);
}
