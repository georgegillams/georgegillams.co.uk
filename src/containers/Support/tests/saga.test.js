import { put, takeLatest } from 'redux-saga/effects';

import { loadLinks, createLink, deleteLink } from '../actions';

import saga, { doLoadLinks, doCreateLink, doDeleteLink } from '../saga';

describe('Support saga', () => {
  let mainSaga;

  beforeEach(() => {
    mainSaga = saga();
  });

  it('Should load links on loadLinks TRIGGER', () => {
    const takeLatestDescriptor = mainSaga.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(loadLinks.TRIGGER, doLoadLinks));
  });

  it('Should create link on createLink TRIGGER', () => {
    mainSaga.next();
    const takeLatestDescriptor = mainSaga.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(createLink.TRIGGER, doCreateLink));
  });

  it('Should delete link on deleteLink TRIGGER', () => {
    mainSaga.next();
    mainSaga.next();
    const takeLatestDescriptor = mainSaga.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(deleteLink.TRIGGER, doDeleteLink));
  });

  describe('loadLinks actions', () => {
    let generator;

    const response = {
      supportMessages: {
        id: 'some-link',
        name: 'link 1',
      },
      status: 200,
    };

    beforeEach(() => {
      generator = doLoadLinks();

      const selectDescriptor = generator.next().value;
      expect(selectDescriptor).toMatchSnapshot();
    });

    it('Should call loadLinks.success on successful API call', () => {
      generator.next();
      const putSuccess = generator.next(response).value;
      generator.next();

      expect(putSuccess).toEqual(put(loadLinks.success(response)));
    });

    it('Should call loadLinks.failure if API returns error', () => {
      const response = {
        error: 'error_type',
        errorMessage: 'Error message',
        status: 500,
      };
      generator.next();
      const putSuccess = generator.next(response).value;
      generator.next();

      expect(putSuccess).toEqual(put(loadLinks.failure(response)));
    });

    it('Should call loadLinks.failure if an exception occurs', () => {
      const response = new Error('Some error');
      const putFailure = generator.throw(response).value;

      expect(putFailure).toEqual(put(loadLinks.failure(response)));
    });
  });

  describe('createLink actions', () => {
    let generator;

    const response = {
      status: 200,
    };

    beforeEach(() => {
      generator = doCreateLink();

      const selectDescriptor = generator.next().value;
      expect(selectDescriptor).toMatchSnapshot();
    });

    it('Should call createLink.success on successful API call', () => {
      generator.next({ linkToCreate: 'some-link' });
      generator.next();
      const putSuccess = generator.next(response).value;
      generator.next();

      expect(putSuccess).toEqual(put(createLink.success(response)));
    });

    it('Should call createLink.failure if API returns error', () => {
      const response = {
        error: 'error_type',
        errorMessage: 'Error message',
        status: 500,
      };
      generator.next({ linkToCreate: 'some-link' });
      generator.next();
      const putSuccess = generator.next(response).value;
      generator.next();

      expect(putSuccess).toEqual(put(createLink.failure(response)));
    });

    it('Should call createLink.failure if an exception occurs', () => {
      generator.next({ linkToCreate: 'some-link' });
      const response = new Error('Some error');
      const putFailure = generator.throw(response).value;

      expect(putFailure).toEqual(put(createLink.failure(response)));
    });
  });

  describe('deleteLink actions', () => {
    let generator;

    const response = {
      status: 200,
    };

    beforeEach(() => {
      generator = doDeleteLink();

      const selectDescriptor = generator.next().value;
      expect(selectDescriptor).toMatchSnapshot();
    });

    it('Should call deleteLink.success on successful API call', () => {
      generator.next({ linkToDelete: 'some-link' });
      generator.next();
      const putSuccess = generator.next(response).value;
      generator.next();

      expect(putSuccess).toEqual(put(deleteLink.success(response)));
    });

    it('Should call deleteLink.failure if API returns error', () => {
      const response = {
        error: 'error_type',
        errorMessage: 'Error message',
        status: 500,
      };
      generator.next({ linkToDelete: 'some-link' });
      generator.next();
      const putSuccess = generator.next(response).value;
      generator.next();

      expect(putSuccess).toEqual(put(deleteLink.failure(response)));
    });

    it('Should call deleteLink.failure if an exception occurs', () => {
      generator.next({ linkToDelete: 'some-link' });
      const response = new Error('Some error');
      const putFailure = generator.throw(response).value;

      expect(putFailure).toEqual(put(deleteLink.failure(response)));
    });
  });
});
