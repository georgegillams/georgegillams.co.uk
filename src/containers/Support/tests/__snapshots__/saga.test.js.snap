// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Support saga createLink actions Should call createLink.failure if API returns error 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`Support saga createLink actions Should call createLink.failure if an exception occurs 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`Support saga createLink actions Should call createLink.success on successful API call 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`Support saga deleteLink actions Should call deleteLink.failure if API returns error 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`Support saga deleteLink actions Should call deleteLink.failure if an exception occurs 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`Support saga deleteLink actions Should call deleteLink.success on successful API call 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`Support saga loadLinks actions Should call loadLinks.failure if API returns error 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "action": Object {
      "type": "app/support/LOAD_LINKS/REQUEST",
    },
    "channel": undefined,
  },
  "type": "PUT",
}
`;

exports[`Support saga loadLinks actions Should call loadLinks.failure if an exception occurs 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "action": Object {
      "type": "app/support/LOAD_LINKS/REQUEST",
    },
    "channel": undefined,
  },
  "type": "PUT",
}
`;

exports[`Support saga loadLinks actions Should call loadLinks.success on successful API call 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "action": Object {
      "type": "app/support/LOAD_LINKS/REQUEST",
    },
    "channel": undefined,
  },
  "type": "PUT",
}
`;
